### ISSUE TITLE
Layout/MultilineMethodCallIndentation false positive for method chains

### ISSUE THREAD

**Issue by poulleas** [rubocop/rubocop#15054]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

False positive

### Cop name

Layout/MultilineMethodCallIndentation

### OK in 1.84.0
```
def my_first_issue(alpha, beta)
  {
    alpha: alpha,
    beta: beta,
    alpha_beta_count: @fetch_beta_handler.retrieve_all_by(alpha: alpha)
                                         .authorized.count,
    beta_count: @fetch_beta_handler.retrieve_all.authorized.count
  }
end

def my_second_issue(id)
  Response::Document.new(
    data: Response::Resource.new(
      data: DataProvider
              .compute_availability_handler
              .compute_availability_for_analysis_context(id, action_kind: ACTION),
      links: NavigationService.build_entity_links(id)
    )
  )
end
```

### Considered OK in 1.86.0 and previous examples auto-corrected
```
def my_first_issue(alpha, beta)
  {
    alpha: alpha,
    beta: beta,
    alpha_beta_count: @fetch_beta_handler.retrieve_all_by(alpha: alpha)
                      .authorized.count,
    beta_count: @fetch_beta_handler.retrieve_all.authorized.count
  }
end

def my_second_issue(id)
  Response::Document.new(
    data: Response::Resource.new(
      data: DataProvider
            .compute_availability_handler
            .compute_availability_for_analysis_context(id, action_kind: ACTION),
      links: NavigationService.build_entity_links(id)
    )
  )
end
```

### Expected behavior

I would expect rubocop not triggered on this code: 
```
def my_first_issue(alpha, beta)
  {
    alpha: alpha,
    beta: beta,
    alpha_beta_count: @fetch_beta_handler.retrieve_all_by(alpha: alpha)
                                         .authorized.count,
    beta_count: @fetch_beta_handler.retrieve_all.authorized.count
  }
end

def my_second_issue(id)
  Response::Document.new(
    data: Response::Resource.new(
      data: DataProvider
              .compute_availability_handler
              .compute_availability_for_analysis_context(id, action_kind: ACTION),
      links: NavigationService.build_entity_links(id)
    )
  )
end
```

### Actual behavior

Rubocop detects offenses:
```
Offenses:

issue_examples.rb:8:42: C: [Corrected] Layout/MultilineMethodCallIndentation: Align .authorized with @fetch_beta_handler.retrieve_all_by(alpha: alpha) on line 7.
                                         .authorized.count,
                                         ^^^^^^^^^^^
issue_examples.rb:17:15: C: [Corrected] Layout/MultilineMethodCallIndentation: Align .compute_availability_handler with DataProvider on line 16.
              .compute_availability_handler
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
issue_examples.rb:18:15: C: [Corrected] Layout/MultilineMethodCallIndentation: Align .compute_availability_for_analysis_context with DataProvider on line 16.
              .compute_availability_for_analysis_context(id, action_kind: ACTION),
```
 and the auto-correct is the following result
```
def my_first_issue(alpha, beta)
  {
    alpha: alpha,
    beta: beta,
    alpha_beta_count: @fetch_beta_handler.retrieve_all_by(alpha: alpha)
                      .authorized.count,
    beta_count: @fetch_beta_handler.retrieve_all.authorized.count
  }
end

def my_second_issue(id)
  Response::Document.new(
    data: Response::Resource.new(
      data: DataProvider
            .compute_availability_handler
            .compute_availability_for_analysis_context(id, action_kind: ACTION),
      links: NavigationService.build_entity_links(id)
    )
  )
end
```

### RuboCop configuration

```yaml

```

### RuboCop version

```shell
1.86.0 (using Parser 3.3.10.2, Prism 1.9.0, rubocop-ast 1.49.1, analyzing as Ruby 4.0, running on ruby 4.0.2) [x86_64-linux]
  - rubocop-factory_bot 2.28.0
  - rubocop-performance 1.26.1
  - rubocop-rspec 3.9.0
```

**Comment by wendelscardua** [comment]
To make things more complicated, the PR ( https://github.com/rubocop/rubocop/pull/14936 ) that was supposed to fix the previous instance of this issue ( https://github.com/rubocop/rubocop/issues/14916 ) actually introduced specs to make this the expected behavior instead of fixing it.

**Comment by fsmanuel** [comment]
Here is another examples that fails after 1.84.0

```rb
class Resource
    filter :season, apply: lambda { |records, values, _options|
      return records if values.blank?

      records.joins(:match)
             .where(matches: { season_id: values })
    }
end
```

After auto-corrected

```rb
class Resource
    filter :season, apply: lambda { |records, values, _options|
      return records if values.blank?

      records.joins(:match)
      .where(matches: { season_id: values })
    }
end
```
